### DISCUSSION TITLE
Prevent server side data fetching of pre-rendered page

### DISCUSSION THREAD

**Discussion by dan-caldwell** [vikejs/vike#1531] [Discussion] [answered]
I have a page with a `/+data.ts` file that looks like this:

```typescript
import axios from 'axios';

export { data }

async function data() {
  try {
    const { data } = await axios.get(myURLHere);
    console.log('fetched data from axios');
    return data;
  } catch (err) {
    console.error(err);
    return {};
  }
}
```

If I have prerendering enabled and do a Vite preview, then navigate to the generated .html file in my `/dist` folder, I see all of the data is in the HTML and in a JSON object.

However, when I load the page, Axios is still fetching the data from the server. Is there any way to prevent this API call if the data is already in the prerendered HTML page?

For reference, my `+Page` looks something like this:

```typescript
export default function Page() {
  const data = useData();
  const headers = data?.headers || [];
  const rows = data?.rows || [];

  return (
    <Table
      headers={headers}
      rows={rows}
    />
  );
}

```

This occurs when running this npm script:

`"preview": "vite build && vite preview"`

I am not currently using the `vike-react` package.

**Comment by brillout** [comment]
Seems to be a user land issue, not a Vike one. That said, minimal reproduction welcome and I'll have a look at it.

**Reply by dan-caldwell** [reply, marked as answer]
I was able to resolve this issue. I wanted some pages to be pre-rendered while others statically generated. I needed to follow the steps here: https://vike.dev/prerender

Vite config:

```typescript
vike({
  prerender: {
    partial: true,
  }
}),
```
And then in the folder of the pages I wanted SSR, I added this `+config.ts` file:

```typescript
export default {
  prerender: false,
}
```

My project is set up in the same way as the `react-streaming` example: https://github.com/vikejs/vike/tree/main/examples/react-streaming

I now run the server via the `npm run server:prod` script.

Everything works great. Thank you for this great package, I have found the control and ease of use to be excellent.

**Reply by brillout** [reply]
Thanks for circling back on this and I'm glad you're enjoying Vike. Is [sponsoring](https://github.com/vikejs/vike/issues/1350) something your company would be up for? It makes a massive difference for the project.
